chapters/en/slides/module5/module5_04.Rmd

```diff
@@ -0,0 +1,348 @@
+    shape='Origin')
+```
+
+Notes:
+Although many of the color schemes in Altair 
+are designed according to the guidelines for effective color uses,
+they can sometimes be difficult to interpret 
+for people with color vision deficiences.
+
+Esppecially color schemes that mix red and green
+as the one in the previous slide.
+
+In addition to using a more suitable color scheme (such as the default),
+we could also change the shape of the points for each category.
+
+Although this is technically redundant
+since the color is already used for the categorical groups,
+it can make your visualization more effective.
+
+For line plots we could achieve a similar effect via the `strokeDash` encoding
+instead of `shape`,
+which together with direct labeling from the previous slide deck
+can facilitate interpretation of these plots.
+
+---
+
+## Specifying custom colors
+
+```{python}
+colors = ['coral', '#4682b4', 'rebeccapurple']
+alt.Chart(cars).mark_point(size=70, filled=True).encode(
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.Color('Origin', title=None, scale=alt.Scale(range=colors)))
+```
+
+Notes:
+Most of the time it is a good idea to stick to the predefined color scheme
+because of the advantages mentioned in the previous slide.
+
+However,
+sometimes the categories we are representing might have a color already associated with them,
+such as political parties or sports team.
+
+In these cases,
+it is often better to design a custom color scheme
+using the colors naturally associated with each category.
+
+In Altair we can create a custom color scheme by passing a list of colors 
+to the `range` parameter of `alt.Scale`.
+
+You can specify colors either by their HTML/CSS name (such as 'coral')
+or their hex code (such as '#4682b4').
+[All HTML/CSS color names can be found in the image in this post](https://stackoverflow.com/a/37232760/2166823).
+
+Hex codes are defines over the range `#000000` for black ("zero color")
+to `#ffffff` for white ("full color")
+and [are easiest chosen via color pickers such as this one](https://colorpicker.me/).
+
+---
+
+## Don't use more than 5-8 distinct hues for categories
+
+<center><img src=/module5/hue-guidelines.png></img></center>
+
+Notes:
+Although many of the categorical color schemes contain ten or more hues,
+if is often not a good idea to use all that many.
+
+The guidelines on what is too many hues differ between different sources
+and also depends on your use case.
+A good rule of thumb is that when you get to around five different hues,
+you should really consider if this is the best way to represent your data
+or if you could split it up into multiple visualizations instead.
+
+If the data is neatly organized in well-separated clusters,
+it is possible that you could visualize more than five colors
+(maybe even ten) effectively,
+but in data where the datapoints are more mixed,
+you will rarely, if ever, be able to go this high.
```

Comment:
You should explain why here - because we start to become unable to distinguish them! You hint at this, but do not explicitly say this. You should!